p2p/src/sync/mod.rs

```diff
@@ -96,6 +97,10 @@ pub struct BlockSyncManager<T: NetworkingService> {
     /// Subsystem handle to Chainstate
     chainstate_handle: subsystem::Handle<Box<dyn chainstate_interface::ChainstateInterface>>,
 
+    /// Subsystem handle to Mempool
+    #[allow(unused)]
+    mempool_handle: subsystem::Handle<Box<dyn MempoolInterface<DummyMempoolChainState>>>,
+
```

Comment:
Is there a reason why syncing is given a handle to mempool?

Reply:
This doesn't seem to be appropriate. Please just pass the mempool handle to p2p, and let the p2p developer manage that internally. I'll make another comment on where to start removing.

Reply:
Done.